packages/core/src/storage/report-gap.test.ts.md

```diff
@@ -0,0 +1,45 @@
+# Snapshot report for `src/storage/report-gap.test.ts`
+
+The actual snapshot is saved in `report-gap.test.ts.snap`.
+
+Generated by [AVA](https://avajs.dev).
+
+## insert var without shrink gap
+
+> Snapshot 1
+
+    `file.sol:1: Layout changed for \`__gap\` (uint256[48] -> uint256[48])␊
+      - Slot changed from 2 to 3`
+
+## delete var and expand gap
+
+> Snapshot 1
+
+    `StorageUpgrade_Gap_V1: Deleted \`b\`␊
+      > Keep the variable even if unused␊
+    ␊
+    file.sol:1: Upgraded \`__gap\` to an incompatible type␊
+      - Bad storage gap resize from 48 to 49␊
+        Size cannot increase here`
+
+## shrink gap without adding var
+
+> Snapshot 1
+
+    `file.sol:1: Upgraded \`__gap\` to an incompatible type␊
+      - Bad storage gap resize from 48 to 47␊
+        Size decrease must match with corresponding variable inserts`
+
+## insert var and shrink gap too much
+
+> Snapshot 1
+
+    `file.sol:1: Upgraded \`__gap\` to an incompatible type␊
+      - Bad storage gap resize from 48 to 46␊
+        Size decrease must match with corresponding variable inserts`
+
+## insert vars and shrink gap not enough
+
+> Snapshot 1
+
+    'file.sol:1: Replaced `__gap` with `d` of incompatible type'
```

Comment:
I find it weird that we're seeing this error here. Makes me think we should give lower cost to `typechange`.